File: android/app/src/main/java/com/vernu/sms/models/SMSPayload.java
package com.vernu.sms.models;

public class SMSPayload {

    private String[] recipients;
    private String message;
    private String smsId;
    private String smsBatchId;
    private Integer simSubscriptionId;

    // Legacy fields that are no longer used
    private String[] receivers;
    private String smsBody;

    public SMSPayload() {
    }

    public String[] getRecipients() {
        return recipients;
    }

    public void setRecipients(String[] recipients) {
        this.recipients = recipients;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public String getSmsId() {
        return smsId;
    }

    public void setSmsId(String smsId) {
        this.smsId = smsId;
    }

    public String getSmsBatchId() {
        return smsBatchId;
    }

    public void setSmsBatchId(String smsBatchId) {
        this.smsBatchId = smsBatchId;
    }

    public Integer getSimSubscriptionId() {
        return simSubscriptionId;
    }

    public void setSimSubscriptionId(Integer simSubscriptionId) {
        this.simSubscriptionId = simSubscriptionId;
    }
}

File: android/app/src/main/java/com/vernu/sms/models/SMSPayload.kt
package com.vernu.sms.models

class SMSPayload {
    var recipients: Array<String>? = null
    var message: String? = null
    var smsId: String? = null
    var smsBatchId: String? = null
    var simSubscriptionId: Int? = null

    // Legacy fields — no longer actively used but kept for backward compatibility
    var receivers: Array<String>? = null
    var smsBody: String? = null
}
